python/wtte/wtte.py

```diff
@@ -166,13 +166,13 @@ def loglik_discrete(y, u, a, b, epsilon=K.epsilon()):
     hazard1 = K.pow((y + 1.0) / a, b)
 
     loglikelihoods = u * \
-        K.log(K.exp(hazard1 - hazard0) - (1.0 - epsilon)) - hazard1
+        K.log((1.0 + epsilon) - K.exp(hazard0 - hazard1)) - hazard0
     return loglikelihoods
```

Comment:
In the discrete case it’s equivalent. It’s just a matter of where we put epsilon:
```
 K.log((1.0 + epsilon) - K.exp(hazard0 - hazard1)) - hazard0
=K.log((1.0 + epsilon) - K.exp(hazard0)/K.exp(hazard1)) - hazard0
=K.log(K.exp(hazard0)[(1.0 + epsilon)/K.exp(hazard0) - 1/K.exp(hazard1)]) - hazard0
=K.log([(1.0 + epsilon)*K.exp(-hazard0) - K.exp(-hazard1)])
=K.log(K.exp(-hazard1)[(1.0 + epsilon)*K.exp(hazard1-hazard0) - 1])
=K.log([(1.0 + epsilon)*K.exp(hazard1-hazard0) - 1])-hazard1
=K.log([(1.0)*K.exp(hazard1-hazard0) - 1])-hazard1
~K.log([K.exp(hazard1-hazard0) - 1])-hazard1
```

Which form is right is just a matter of style. My reasons is that I like to emphasis that for positive distributions discretized loglikelihood can always be written on the form `u*log[exp(Λ(y+1)-Λ(y))-1]-Λ(y+1)`, which is a form I chose because empirically it seemed most numerically stable/efficient and that automatic differentiation seemed to unfold it efficiently. I may be wrong about this however so please provide me with counterexamples if you have them. 
See proposition [2.26](http://publications.lib.chalmers.se/records/fulltext/253611/253611.pdf).

Reply:
Yeah, the only true difference was in case of `u=0`.